src/components/ChannelsAvailability/ChannelsAvailabilityWithPermission.tsx

```diff
@@ -0,0 +1,25 @@
+import { hasPermission } from "@saleor/auth/misc";
+import ChannelsAvailability, {
+  ChannelsAvailabilityProps
+} from "@saleor/components/ChannelsAvailability/ChannelsAvailability";
+import useUser from "@saleor/hooks/useUser";
+import { PermissionEnum } from "@saleor/types/globalTypes";
+import React from "react";
+
+export const ChannelsAvailabilityWithPermission: React.FC<Omit<
+  ChannelsAvailabilityProps,
+  "hasManageChannelsPermission"
+>> = props => {
+  const { user } = useUser();
+  const hasManageChannelsPermission =
+    !!user && hasPermission(PermissionEnum.MANAGE_CHANNELS, user);
+
+  return (
+    <ChannelsAvailability
+      {...props}
+      hasManageChannelsPermission={hasManageChannelsPermission}
```

Comment:
`ChannelsAvailability` doesn't need to know anything about permissions, I would change it to something like: 
```suggestion
      manageChannels={hasManageChannelsPermission}
```
as the `ChannelsAvailability` component would only care about whether managing channels UI within it should be displayed, while a decision on when it should be displayed, e.g. when permission is available, would be left for upper layer components.